src/win32/window_manager.cc

```diff
@@ -0,0 +1,44 @@
+#include "../window_manager.h"
+#include <windows.h>
+
+BOOL CALLBACK EnumWindowsProc(HWND hwnd, LPARAM lparam) {
+    std::vector<WindowHandle>* windowHandles = reinterpret_cast<std::vector<WindowHandle>*>(lparam);
+    if (windowHandles != nullptr) {
+        windowHandles->push_back(reinterpret_cast<WindowHandle>(hwnd));
+    }
+    return TRUE;
+}
+
+std::vector<WindowHandle> getWindows() {
+    std::vector<WindowHandle> windowHandles;
+
+    EnumWindows (&EnumWindowsProc, reinterpret_cast<LPARAM>(&windowHandles));
+
+    return windowHandles;
+}
+
+WindowHandle getActiveWindow() {
+    return reinterpret_cast<WindowHandle>(GetForegroundWindow());
```

Comment:
Considering the [documentation of GetForegroundWindow](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getforegroundwindow), it could return `NULL` in some circumstances. It might make sense to handle this case instead of returning a hard cast of `NULL` to a `WindowHandle`.

Reply:
According to the defintion of `NULL`

```c
#ifndef NULL
    #ifdef __cplusplus
        #define NULL 0
```

I think it's fine to treat it like a `WindowHandle`

Reply:
Okay, so casting `NULL` would be resolved into a `WindowHandle` for window 0. Given you have 3 open windows and you call `getActiveWindow` while the third is active. Due to *some circumstances*, `NULL`  would be returned. A follow up call of `getWindowRect` for example would return the rect of window 1 instead of windows 3 without any information to the user, wouldn't it?

Reply:
Yeah, you're right with that one. I addressed it.